test/fw/bin/pbs_snapshot

```diff
@@ -267,25 +268,46 @@ if __name__ == '__main__':
             map_file = os.path.join(out_abspath, "obfuscate.map")
 
     if additional_hosts is not None:
-        du = DshUtils()
+        # Capture snapshot of remote hosts in addition to the main host
 
-        # Run child pbs_snapshot commands on those hosts
         hostnames = additional_hosts.split(",")
         childsnap_threads = {}
         for host in hostnames:
-            thread = Thread(target=childsnap_thread, args=(
+            thread = Thread(target=remotesnap_thread, args=(
                 ptl_logger, host))
             thread.start()
             childsnap_threads[host] = thread
 
-        # Capture snapshot on the main host in the meantime
-        with PBSSnapUtils(out_dir, primary_host=primary_host,
-                          acct_logs=acct_logs,
-                          daemon_logs=daemon_logs, map_file=map_file,
-                          anonymize=anonymize, create_tar=False,
-                          log_path=log_path,
-                          with_sudo=with_sudo) as snap_utils:
-            main_snap = snap_utils.capture_all()
+        # Capture snapshot of the main host in the meantime
+        thread_p = None
+        if not du.is_localhost(primary_host):
+            # The main host is remote
+            thread_p = Thread(target=remotesnap_thread,
```

Comment:
We have 3 distinct cases:
- when there are additional hosts, we want to capture the main snapshot as a directory and put all child snapshots in it, then tar it up.
- when we have just a remote primary host, we have to run pbs_snapshot on the remote host and capture a tar
- when we have just the local host to snap, we capture it via PBSSnapUtils as a tar

So, all the checks that I have right now will still exist even if I add the primary to the list of all hosts to capture as we need to know which case we fall into and do something special for that case. I decided to not do what you are suggesting because I think that would have made the code difficult to follow.

Reply:
Fair enough
I have no more comments